src/jit/morph.cpp

```diff
@@ -7671,12 +7671,13 @@ void Compiler::fgMorphTailCall(GenTreeCall* call)
         call->gtFlags |= GTF_EXCEPT;
     }
 
-    // Now inject a placeholder for the real call target that codegen
-    // will generate
-    GenTreePtr arg = new (this, GT_NOP) GenTreeOp(GT_NOP, TYP_I_IMPL);
+    // Now inject a placeholder for the real call target that codegen will generate
 #ifdef LEGACY_BACKEND
+    GenTreePtr arg = new (this, GT_NOP) GenTreeOp(GT_NOP, TYP_I_IMPL);
     codeGen->genMarkTreeInReg(arg, REG_TAILCALL_ADDR);
-#endif
+#else // !LEGACY_BACKEND
+    GenTreePtr arg = gtNewIconNode(0, TYP_I_IMPL);
+#endif // !LEGACY_BACKEND
```

Comment:
I made these changes, and the ones below, by adding `#ifdef` to avoid changing any `LEGACY_BACKEND` behavior, but to make the code do just what the amd64 code does.